gems/sorbet-runtime/lib/types/props/utils.rb

```diff
@@ -23,16 +23,10 @@ def self.deep_clone_object(what, freeze: false)
       h
     when Regexp
       what.dup
+    when T::Enum
+      what
     else
-      # Some unfortunate nastiness to get around Opus::Enum potentially not
-      # being defined.
-      if defined?(Opus) && defined?(Opus::Enum) && what.class == Opus::Enum
-        what
-      elsif defined?(T::Enum) && what.class == T::Enum
-        what
-      else
-        what.clone
-      end
```

Comment:
We can skip these `defined?` checks because it's defined in sorbet-runtime now and thus always defined.